pkg/templates/danglingservicemonitor/template_test.go

```diff
@@ -0,0 +1,187 @@
+package danglingservicemonitor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext/mocks"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/danglingservicemonitor/internal/params"
+
+	k8sMonitoring "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	service1           = "service1"
+	service2           = "service2"
+	servicemonitor1    = "servicemonitor-matches-service1"
+	servicemonitor2    = "servicemonitor-matches-service2"
+	servicemonitorNone = "servicemonitor-matches-none"
+)
+
+var emptyLabelSelector = metaV1.LabelSelector{
+	MatchLabels: map[string]string{},
+}
+
+var labelselector1 = metaV1.LabelSelector{
+	MatchLabels: map[string]string{"app": "service1-test"},
+}
+
+var labelselector2 = metaV1.LabelSelector{
+	MatchLabels: map[string]string{"app": "service2-test"},
+}
+
+var invalidlabelselector = metaV1.LabelSelector{
+	MatchLabels: map[string]string{"-incorrect-labelselector-and-it-is-too-long-for-kubernetes-today-": "test"},
+}
+var namespaceselector1 = k8sMonitoring.NamespaceSelector{
+	MatchNames: []string{"test1"},
+}
+
+var namespace1 = "test1"
+
+var namespace2 = "test2"
+
+func TestDanglingServiceMonitor(t *testing.T) {
+	suite.Run(t, new(DanglingServiceMonitorTestSuite))
+}
+
+type DanglingServiceMonitorTestSuite struct {
+	templates.TemplateTestSuite
+
+	ctx *mocks.MockLintContext
+}
+
+func (s *DanglingServiceMonitorTestSuite) SetupTest() {
+	s.Init("dangling-servicemonitor")
+	s.ctx = mocks.NewMockContext()
+}
+
+func (s *DanglingServiceMonitorTestSuite) AddServiceMonitorWithLabelSelector(name string, labelSelector metaV1.LabelSelector) {
+	s.ctx.AddMockServiceMonitor(s.T(), name)
+	s.ctx.ModifyServiceMonitor(s.T(), name, func(servicemonitor *k8sMonitoring.ServiceMonitor) {
+		servicemonitor.Spec.Selector = labelSelector
+		fmt.Println(servicemonitor)
```

Comment:
```suggestion
```
Why print?

Reply:
think this was an old bit for debugging, thanks for the catch